src/envoy/alts/alts_socket_factory.cc

```diff
@@ -31,30 +31,27 @@ namespace Configuration {
 
 using ::google::protobuf::RepeatedPtrField;
 
-// Returns true if the peer's service account is found in targets, otherwise
+// Returns true if the peer's service account is found in peers, otherwise
 // returns false and fills out err with an error message.
 static bool doValidate(const tsi_peer &peer,
-                       const RepeatedPtrField<std::string> &targets,
+                       std::unordered_set<std::string> peers,
                        std::string &err) {
   for (size_t i = 0; i < peer.property_count; ++i) {
     std::string name = std::string(peer.properties[i].name);
     std::string value = std::string(peer.properties[i].value.data,
                                     peer.properties[i].value.length);
-    if (name.compare(TSI_ALTS_SERVICE_ACCOUNT_PEER_PROPERTY) == 0) {
-      for (int j = 0; j < targets.size(); ++j) {
-        if (value.compare(targets[j]) == 0) {
-          return true;
-        }
-      }
+    if (name.compare(TSI_ALTS_SERVICE_ACCOUNT_PEER_PROPERTY) == 0 &&
+        peers.count(value) > 0) {
+      return true;
     }
   }
 
-  err = "Couldn't find peer's service account in targets: ";
-  for (int i = 0; i < targets.size(); ++i) {
-    if (i != 0) {
+  err = "Couldn't find peer's service account in peer_service_accounts: ";
+  for (auto it = peers.cbegin(); it != peers.cend(); ++it) {
```

Comment:
Use absl::StrJoin?

Reply:
Thanks, that's much better for now. I didn't know that before.